src/Task.ts

```diff
@@ -353,7 +353,8 @@ export class Task {
         // Write next occurrence before previous occurrence.
         newTasks.push(toggledTask);
 
-        return newTasks;
+        const { recurrenceOnNextLine: recurrenceOnNextLine } = getSettings();
+        return recurrenceOnNextLine ? newTasks.reverse() : newTasks;
```

Comment:
PR #1925 adds a new method to Task -  `toggleUpdate()`.

I do not yet understand why - but I would like to understand if it will conflict this.

Reply:
In particular, if a user changes their setting to switch the order of adding new recurrence, what code will be broken.

Reply:
There is code all over Tasks that assumes that the new recurrence is the first task returned.

It's clearly documented in this method that that is the behaviour.

I am afraid changing this behaviour is a very high risk change. And it will mean that every place in future where this method is called will have to check the settings to know which is the old task and which is the new.

I think the correct place for the reversing of tasks is the point where the toggled lines are written to the note, so that this PR's change is much more localised.

Reply:
It won't. This PR changes `Task.toggle()`, which is called by `ToggleDone.toggleLine()` (current state) and the `Task.toggleUpdate()` in the PR #1925 : 
```
    public toggleUpdate() {
         const newTasks = this.toggle();
```

Reply:
I guess now the `Task.toggleWithRecurrenceInUsersOrder()` has to be used there if my PR is merged first or I will update the call in the other case.